components/form/builder/src/index.js

```diff
@@ -0,0 +1,357 @@
+        size={size}
+        placeholder={placeholder}
+        onChange={value => this._handleInputChange(field, value)}
+      />
+    )
+  }
+
+  /**
+   * Renders the component corresponding to the received field name
+   * @param {Object} props
+   */
+  _renderFormFieldItem = ({props}) => {
+    const {type} = props
+    switch (type) {
+      case 'select':
+        return this._renderSelect(props)
+      case 'button-group':
+        return this._renderButtonGroup(props)
+      case 'input':
+        return this._renderInput(props)
+      case 'text-area':
+        return this._renderTextArea(props)
+    }
+  }
+
+  /**
+   * Renders the form field corresponding to the received field name
+   * @param {string} field
+   * @param {number} index
+   */
+  _renderFormField = (field, index) => {
+    const capitalizedKey = `${field[0].toUpperCase()}${field.slice(1)}` // SUI-JS ??
+    const disabled = !this.state[field].length
+    const id = `${BASE_CLASS}-${field}`
+    const props = {field, id, disabled, ...this._config[field], BASE_CLASS}
+    const fieldClass = cx(
+      `${FORM_ITEM_CLASS} ${FORM_ITEM_CLASS}${capitalizedKey}`,
+      {
+        [FORM_ITEM_DISABLED_CLASS]: disabled
+      }
+    )
+    return (
+      <div key={index} className={fieldClass}>
+        {this._renderFormFieldItem({props})}
+      </div>
+    )
+  }
+
+  /**
+   * Execute the form submit
+   * @param {Event} e
+   */
+  _handleSubmit = e => {
+    e.preventDefault()
+    this.props.onSubmit(this.state)
+  }
+
+  /**
+   * Render method that maps config KEYS and generates form fields dynamically depending on the config, a specific type of form field (select, button group, text-area, etc.)
+   */
+  render() {
+    return (
+      <div className={BASE_CLASS}>
+        <form className={FORM_WRAP_CLASS} onSubmit={this._handleSubmit}>
+          {this._formFields.map((field, index) =>
+            this._renderFormField(field, index)
+          )}
+          <AtomButtom isSubmit>{this._submitText}</AtomButtom>
+        </form>
+      </div>
+    )
+  }
+}
+
+FormBuilder.displayName = 'FormBuilder'
+
+FormBuilder.propTypes = {
+  /** Config with the form specifications */
+  config: PropTypes.shape({
+    /** Refers to the formfield Type */
+    type: PropTypes.string.isRequired,
+    /** Refers to the following field that the formbuilder must request */
+    next: PropTypes.string,
+    /** Refers to the formfield placeholder */
+    placeholder: PropTypes.string,
+    /** Refers to the formfield label */
+    label: PropTypes.string,
+    /** Refers to the formfield errorText */
+    errorText: PropTypes.string,
+    /** Refers to whether formField is persistent */
+    persists: PropTypes.bool,
+    /** Refers to the formfield input type */
+    inputType: PropTypes.string,
+    /** Refers to the formfield size */
+    size: PropTypes.string
+  }),
+  /** Function that requests the start listings of the form */
```

Comment:
Minor suggestion: `func executed on component load. May be used to intialize form data.`

Reply:
DONE!